frontend/pages/SoftwarePage/SoftwareVulnerabilities/SoftwareVulnerabilities.tsx

```diff
@@ -66,11 +89,156 @@ const SoftwareVulnerabilities = ({
+    IGetVulnerabilityQueryKey[]
+  >(
+    [
+      {
+        scope: "softwareVulnByCVE",
+        vulnerability: (query && stripQuotes(query)) || "",
+        teamId,
+      },
+    ],
+    ({ queryKey }) => {
+      // Revisit: Refactor to return status alongside data and check for 204 instead of !data
+      return softwareVulnAPI.getVulnerability(queryKey[0]);
+    },
+    {
+      ...DEFAULT_USE_QUERY_OPTIONS,
+      retry: false,
+      onSuccess: (data) => {
+        // Handle 204 response which doesn't return data if it is a known CVE but doesn't exist in response
+        if (!data) {
+          setTableData({
+            count: 0,
+            counts_updated_at: "",
+            vulnerabilities: [],
+            meta: {
+              has_next_results: false,
+              has_previous_results: false,
+            },
+          });
+          setEmptyStateReason("known-vuln");
+        }
+        // If filtering for exploited vulns, hide vulnerability if cisa_known_exploit is false
+        else if (
+          queryParams.exploit &&
+          !data.vulnerability.cisa_known_exploit
+        ) {
+          setTableData({
+            count: 0,
+            counts_updated_at: "",
+            vulnerabilities: [],
+            meta: {
+              has_next_results: false,
+              has_previous_results: false,
+            },
+          });
+          setEmptyStateReason("no-matching-items");
+          // Otherwise return IVulnerabilityResponse as IVulnerabilitiesResponse format
+        } else {
+          setTableData({
+            count: 1,
+            counts_updated_at: data.vulnerability.hosts_count_updated_at,
+            vulnerabilities: [data.vulnerability],
+            meta: {
+              has_next_results: false,
+              has_previous_results: false,
+            },
+          });
+        }
+      },
+      onError: (err) => {
+        // Type assertion for failing "Property 'data' does not exist on type 'AxiosError<IApiError, any>"
+        const error = err as AxiosError<IApiError> & {
+          data?: IApiError;
+        };
+
+        // Handle 400 response which is an invalid CVE format
+        if (error.status === 400) {
+          if (
+            error?.data?.errors &&
+            error.data.errors[0].reason.includes(
+              "That vulnerability (CVE) is not valid."
+            )
+          ) {
+            setTableData({
+              count: 0,
+              counts_updated_at: "",
+              vulnerabilities: [],
+              meta: {
+                has_next_results: false,
+                has_previous_results: false,
+              },
+            });
+            setEmptyStateReason("invalid-cve");
+          }
+
+          // Handle 404 response which is BE validated CVE string but not a known CVE
+        } else if (error.status === 404) {
+          if (
+            error?.data?.errors &&
+            (error.data.errors[0].reason.includes("This is not known CVE.") ||
```

Comment:
Should this be `...not a known...`?

Reply:
I matched `vulnerabilities.go`, we can update both in a follow up.

Reply:
follow up PR here: https://github.com/fleetdm/fleet/pull/21590